src/app/item-page/simple/field-components/file-section/file-section.component.html

```diff
@@ -14,7 +15,21 @@
             @if (!last) {
               <span innerHTML="{{separator}}"></span>
             }
-          </ds-file-download-link>
+          </ds-item-secure-file-download-link>
+          } @else {
+            <ds-file-download-link [bitstream]="file" [item]="item">
+            <span>
+              @if (primaryBitsreamId === file.id) {
+                <span class="badge bg-primary">{{ 'item.page.bitstreams.primary' | translate }}</span>
+              }
+              {{ dsoNameService.getName(file) }}
+            </span>
+              <span> ({{(file?.sizeBytes) | dsFileSize }})</span>
+              @if (!last) {
+                <span innerHTML="{{separator}}"></span>
+              }
+            </ds-file-download-link>
```

Comment:
The only thing that's different here is the component selector, and it's not likely we'd need to make any other changes at this level. I'd advise to move this "choice" into the link component.

Reply:
I've done this and hopfully restored the file section to its former self